idaes/apps/grid_integration/multiperiod/price_taker_model.py

```diff
@@ -0,0 +1,408 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+
+import pandas as pd
+
+from pyomo.environ import (
+    ConcreteModel,
+    Block,
+    Var,
+    RangeSet,
+    Objective,
+    Constraint,
+    NonNegativeReals,
+    Expression,
+    maximize,
+)
+
+from idaes.apps.grid_integration import MultiPeriodModel
+from idaes.apps.grid_integration.multiperiod.design_and_operation_models import (
+    DesignModelData,
+    OperationModelData,
+)
+
+from sklearn.cluster import KMeans
+from kneed import KneeLocator
+
+import matplotlib.pyplot as plt
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class PriceTakerModel(ConcreteModel):
+    @staticmethod
+    def generate_daily_data(raw_data, day_list):
+
+        daily_data = pd.DataFrame(columns=day_list)
+
+        # Extracting data to populate empty dataframe
+        i = 0
+        j = 24
+        day = 1
+        while j <= len(raw_data):
+            daily_data[day] = raw_data[i:j].reset_index(drop=True)
+            i = j
+            j = j + 24
+            day = day + 1
+
+        return daily_data
+
+    @staticmethod
+    def reconfigure_raw_data(self, raw_data):
+        """
+        Reconfigures the raw price series data into a usable form
+
+        Args:
+            raw_data: imported price series data
+
+        Returns:
+            daily_data: reconfigured price series data
+        """
+        # Get column headings
+        column_head = raw_data.columns.tolist()
+        # Remove the date/time column
+        scenarios = column_head[2:]
+
+        # Creating an empty dataframe to store daily data for clustering
+        day_list = list(range(1, (len(raw_data) // 24) + 1))
+
+        # Generate daily data
+        for i in scenarios[0:1]:
+            daily_data = self.generate_daily_data(
+                raw_data=raw_data[i], day_list=day_list
+            )
+
+        return daily_data
+
+    @staticmethod
+    def get_optimal_n_clusters(daily_data, kmin=None, kmax=None, sample_weight=None):
```

Comment:
Combine `get_optimal_n_clusters` and `get_elbow_plot` methods.

Reply:
Add an argument for `seed` and save it as a hidden argument. Ensure that all methods which use Kmeans clustering code use the same seed value.